run_bot.sh

```
@@ -187,15 +324,32 @@ run_with_config() {
     fi
     print_info "Output directory: $output_dir"
     
+    # Get available ports for this bot instance
+    local port_info
+    port_info=$(get_next_bot_ports)
+    if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)